src/components/media-link/media-link.stories.mdx

```diff
@@ -0,0 +1,82 @@
+import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
+import { mediaArgTypes } from '../../objects/media/demo/arg-types.js';
+import { avatarArgTypes } from '../avatar/demo/arg-types.js';
+import template from './media-link.twig';
+const mediaLinkArgTypes = {
+  ...mediaArgTypes,
+  label: {
+    type: { name: 'string' },
+    description: 'The text content for the Media Link.',
+  },
+  href: {
+    type: { name: 'string' },
+    description: 'The destination URL for the Media Link.',
+    table: {
+      defaultValue: {
+        summary: '#',
+      },
+    },
+  },
+  icon: {
+    type: { name: 'string' },
+    description: 'A key corresponding to one of our compact icons.',
+  },
+  icon_class: {
+    type: { name: 'string' },
+    description:
+      'Optional class name(s) to append to the icon. Useful for adjusting size or color.',
+  },
+  action_class: {
+    type: { name: 'string' },
+    description:
+      'Optional class name(s) to append to the action element (the link itself). Useful for applying typographic adjustments.',
+  },
+  avatar: {
+    ...avatarArgTypes.src,
+    description:
+      'The `src` path for an Avatar component to use as the media object.',
+  },
+  avatar_alt: avatarArgTypes.alt,
+  avatar_width: avatarArgTypes.width,
+  avatar_height: avatarArgTypes.height,
+  avatar_srcset: avatarArgTypes.srcset,
+  avatar_sizes: avatarArgTypes.sizes,
+  avatar_size: avatarArgTypes.size,
+  avatar_shape: avatarArgTypes.shape,
```

Comment:
I resisted the urge to automate this by looping over the `avatarArgTypes` object's keys. The Media Link template can only pass along the properties we've added to the template by hand, so if it felt appropriate for these `argTypes` to be manually connected in a similar fashion.